app/components/bali/page_header/component.rb

```diff
@@ -5,20 +5,39 @@ module PageHeader
     class Component < ApplicationViewComponent
       attr_reader :options
 
-      renders_one :title, ->(text = nil, **options, &block) do
-        options[:class] ||= 'title is-1 mb-0'
-        text.present? ? tag.h1(text, **options) : tag.div(&block)
+      renders_one :title, ->(text = nil, type: :h3, **options, &block) do
+        options = prepend_class_name(options, class_names('title', heading_size_class(type)))
+
+        heading_tag(text, type, **options, &block)
       end
 
-      renders_one :subtitle, ->(text = nil, **options, &block) do
-        options[:class] ||= 'subtitle mt-0 is-6'
-        text.present? ? tag.p(text, **options) : tag.div(&block)
+      renders_one :subtitle, ->(text = nil, type: :h5, **options, &block) do
+        options = prepend_class_name(options, class_names('subtitle', heading_size_class(type)))
+
+        heading_tag(text, type, **options, &block)
       end
 
       def initialize(title: nil, subtitle: nil, **options)
         @title = title
         @subtitle = subtitle
-        @options = prepend_class_name(options, 'page-header-component')
+        @options = prepend_class_name(options, 'page-header-component is-mobile')
+      end
+
+      def heading_size_class(type)
+        {
+          'is-1': type == :h1,
+          'is-2': type == :h2,
+          'is-3': type == :h3,
+          'is-4': type == :h4,
+          'is-5': type == :h5,
+          'is-6': type == :h6
+        }
+      end
+
+      def heading_tag(text, type, **options, &)
+        size = type.to_s.gsub(/^h/, '')
+
+        text.present? ? tag.send("h#{size}", text, **options) : tag.div(&)
```

Comment:
Here probably better to use tag_name as argument to avoid confusion with the tag method.

```suggestion
      def heading_tag(text, tag_name, **options, &)
        text.present? ? tag.send("h#{size}", text, **options) : tag.div(&)
```